core/build/build.odin

```diff
@@ -0,0 +1,304 @@
+package build
+
+import "core:strings"
+import "core:os"
+import "core:path/filepath"
+import "core:sync"
+import "core:encoding/json"
+import "core:log"
+import "core:runtime"
+import "core:fmt"
+import "core:slice"
+
+import "core:c/libc" // TODO: remove dependency on libc after core:os2
+
+@(private="file") _context: runtime.Context
+
+@init
+_ :: proc() {
+	_context = create_default_context()
+}
+
+create_default_context :: proc() -> runtime.Context {
+	context = runtime.default_context()
+	context.allocator = context.temp_allocator
+	context.logger = log.create_console_logger(.Debug, {.Level})
+	return context
+}
+
+default_context :: #force_inline proc() -> runtime.Context {
+	return _context
+}
+
+shell_exec :: proc(cmd: string, echo: bool) -> int {
```

Comment:
Maybe worth adding overloads that take arguments as a slice of strings and go directly to `execve` / `StartProcess` (forgot windows name for it). I'm not always going to be running Odin commands in my build, and sometimes these arguments are dynamic it would be much easier to construct them with dynamic arrays than a string builder

Reply:
Actually thinking about it maybe doing `cmd` as string and passing it directly to libc's `system()` function is not going to get us rid of libc dependency. I've only seen part of what it does (strace outputs for the shells are **huge**), but from what I can guess it creates a thread and runs `execvp`, passing it the shell and the commands as the arguments for that shell. On windows the thread is probably not created.

I'd say we shouldn't depend on a particular shell to perform this functionality so this is how you can remove the libc dependency: If the user passes a single command string, split it up into separate arguments in some well-documented way that is reliable (splitting by space / handling quotation with either single or double quote should be more than enough), then pass that to `execvp` or `CreateProcessA`

Reply:
Noted. I'll get to it today or tomorrow. I'll need to create a vm or use wsl to test things properly, but it shouldnt be too hard to remove libc entirely. 

I'd keep both this and `do` removal as low priority atm, as these are easy to fix issues once the api is finalized and bugs are purged